### PR TITLE
16.0][BUG]Reset password not working properly

### PR DESCRIPTION
**Pull request by Oluwaseun241**
Description of the issue/feature this PR addresses:
Impacted versions:
16.0

Steps to reproduce:
Create new user
Let him do the normal signup
Send Password reset mail

Current behavior before PR:
User will get redirected to login page

Desired behavior after PR is merged:
User would get to the /reset/password site

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Oluwaseun241** [comment]
please review

**Comment by mart-e** [comment]
@Oluwaseun241 thank you for your PR but it seems there is still several issues

- you have not [signed the CLA](https://www.odoo.com/sign-cla)
- you have an indentation issue
- check that all the tests pass [for your branch](https://runbot.odoo.com/runbot/bundle/oluwaseun241-16-0-135206)
- please describe a bit more how to reproduce (a video if you are not sure), I have tested and got the right URL

**Comment by BT-apadron** [comment]
Hello,

I try your solution with one change and works completely fine. You only need to indent ones to the right from line 66 until line 100 in _res_partner.py_.

Also, I recommend you change the title of the PR to something more specific, like **[16.0][BUG]Reset password not working properly**

If you need some help, no doubt to ask.

Best regards,
Alejandro Padrón.

**Comment by Oluwaseun241** [comment]
Okay thanks

File: addons/auth_signup/models/res_partner.py [hunk 1]
```diff
@@ -57,47 +57,51 @@ def _get_signup_url_for_action(self, url=None, action=None, view_type=None, menu
...
+            if res_id:
+                fragment['res_id'] = res_id
+
+            if fragment:
+                query['redirect'] = base + werkzeug.urls.url_encode(fragment)
+
+        signup_url = "/web/%s?%s" % (route, werkzeug.urls.url_encode(query))
+        if not self.env.context.get('relative_url'):
+            signup_url = werkzeug.urls.url_join(base_url, signup_url)
+        res[partner.id] = signup_url
```
Comment (BT-apadron) [inline addons/auth_signup/models/res_partner.py:100]: You must indent ones to the right of all this code selected. The code selected must be inside the `for partner in self:`

**Comment by BT-rfiore** [comment]
Hey @Oluwaseun241  
Has the password reset bug been officially resolved in the odoo core branch?
Can I do a core update to resolve the issue?

Thank you for your help and amazing work!

Kind Regards,

Riccardo

**Comment by Oluwaseun241** [comment]
@BT-rfiore 
No you can go ahead with your own fix
I closed it because it's was failing some test by the bot but the code was perfect 
Seems I need to slow down and know my way around here

**Pull request closed without merging**